### PR TITLE
Docs about the DB interactions in the relayer 

### PR DESCRIPTION
**Pull request by sczembor**
## Description

Closes: #42 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
    
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,63 @@ Mock -> WIP -> alpha -> beta -> production
 | Native-\>Bitcoin | :x:    |
 | Bitcoin SPV      |        |
 
+## Database Interactions
```
Comment (robert-zaremba) [inline README.md:39]: let's move docs to `docs/db.md` and `docs/flow.md` and here only include list of content ("index")

File: README.md [hunk 2]
```diff
@@ -36,6 +36,63 @@ Mock -> WIP -> alpha -> beta -> production
...
+| Column | Data Type | Description |
+|---|---|---|
+| `btc_tx_id` | INTEGER | The ID of the Bitcoin transaction (primary key) |
+| `raw_tx` | BLOB | The raw transaction data as a byte array |
+| `tx_hash` | BLOB | The transaction hash as a byte array |
+| `status` | INTEGER | The status of the transaction (0: Pending, 1: Signed, 2: Broadcasted, 3: Confirmed) |
+
+### Functions
+
+The relayer interacts with the database using the following functions:
```
Comment (robert-zaremba) [inline README.md:56]:
```suggestion
The relayer interacts with the database using the following functions in the `dal` package:
```

**Review by robert-zaremba — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -36,6 +36,63 @@ Mock -> WIP -> alpha -> beta -> production
...
+
+1. `InsertTx()`: Inserts a new transaction into the database.
+2. `GetTx()`: Retrieves a transaction by its ID.
+3. `GetPendingTxs()`: Retrieves all transactions with a "Pending" status.
+4. `UpdateTxStatus()`: Updates the status of a transaction.
+
+### Flow
+
+The following diagram illustrates the flow of a transaction through the relayer and the database. This is one `Relayer` but it interacts with three different networks (`Native`, `IKA` and `Bitcoin`)
```
Comment (vuvoth) [inline README.md:65]:
```suggestion
The following diagram illustrates the flow of a transaction through the relayer and the database. This is one `Relayer`, but it interacts with three different networks (`Native`, `Ika` and `Bitcoin`)
```

File: README.md [hunk 4]
```diff
@@ -36,6 +36,63 @@ Mock -> WIP -> alpha -> beta -> production
...
+
+```mermaid
+flowchart TB
+ subgraph subGraph0["Relayer (Native network)"]
+    direction TB
+        B["Indexes transaction"]
+        A["User submits transaction"]
+  end
+ subgraph subGraph1["Relayer (IKA network)"]
```
Comment (vuvoth) [inline README.md:74]:
```suggestion
 subgraph subGraph1["Relayer (Ika network)"]
```

Comment (vuvoth) [inline README.md:74]:
```suggestion
 subgraph subGraph1["Relayer (Ika network)"]
```

File: README.md [hunk 5]
```diff
@@ -36,6 +36,63 @@ Mock -> WIP -> alpha -> beta -> production
...
+        E["IKA signs transaction"]
```
Comment (vuvoth) [inline README.md:75]:
```suggestion
        E["Ika signs transaction"]
```

File: README.md [hunk 6]
```diff
@@ -36,6 +36,63 @@ Mock -> WIP -> alpha -> beta -> production
...
+        D["Sends to IKA for signing"]
```
Comment (vuvoth) [inline README.md:76]:
```suggestion
        D["Sends to Ika for signing"]
```

**Review by vuvoth — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (sczembor) [reply README.md]: Done

**Pull request merged**